packages/graphql-language-service-parser/src/benchmark/index.ts

```diff
@@ -0,0 +1,151 @@
+    name,
+    mean: `${1.0 / stats.mean} ops / sec`,
+    variance: stats.variance,
+    rme: `${stats.rme}%`,
+    lines: schema.split('\n').length,
+  });
+
+  console.log(`Completed test suite: ${name}`);
+};
+
+const runSplitTest = (name: string, schema: string) => {
+  const stats: IStats[] = [];
+
+  const suite = new Benchmark.Suite();
+  const parser = onlineParser();
+  let state: any = parser.startState();
+
+  schema.split('\n').forEach((line, index) => {
+    let prevState: any;
+    let completeState: any;
+
+    suite.add({
+      maxTime: 0.1,
+      onStart: () => {
+        prevState = { ...state };
+      },
+      fn: () => {
+        const stream = new CharacterStream(line);
+
+        while (!stream.eol()) {
+          parser.token(stream, state);
+          if (state.kind === 'Invalid') {
+            console.log(state.kind, line, index);
+            throw new Error('Invalid');
+          }
+        }
+
+        completeState = state;
+        state = { ...prevState };
+      },
+      onError: console.log,
+      onComplete: (e: any) => {
+        state = completeState;
+        stats.push(e.target.stats);
+      },
+    });
+  });
+
+  console.log(`Started test suite: ${name}`);
+
+  suite.run();
+
+  const results = stats.reduce(
+    (result, stat) => {
+      result.mean += stat.mean / stats.length;
+      result.rme += stat.rme / stats.length;
+      result.variance += stat.variance / stats.length;
+
+      return result;
+    },
+    { mean: 0, variance: 0, rme: 0 },
+  );
+
+  printResult(results, name, schema);
+};
+
+const runWholeTest = (name: string, schema: string) => {
+  const suite = new Benchmark.Suite('', {
+    onComplete: (e: any) => printResult(e.target.stats, name, schema),
+  });
+
+  const parser = onlineParser();
+  const state: any = parser.startState();
+
+  suite.add(() => {
+    const stream = new CharacterStream(schema);
+
+    while (!stream.eol()) {
+      parser.token(stream, state);
+      if (state.kind === 'Invalid') {
+        console.log(state.kind);
+        throw new Error('Invalid');
+      }
+    }
+  });
+
+  console.log(`Started test suite: ${name}`);
+  suite.run();
+};
+
+const runGraphqlParserTest = (name: string, schema: string) => {
+  const suite = new Benchmark.Suite('', {
+    onComplete: (e: any) => printResult(e.target.stats, name, schema),
+  });
+
+  suite.add({
+    fn() {
+      try {
+        parse(schema);
+      } catch (err) {
+        console.log(err);
+        throw err;
+      }
+    },
+  });
+
+  console.log(`Started test suite: ${name}`);
+  suite.run();
+};
+
+const kitchenSchema = fs.readFileSync(
+  path.resolve(__dirname, './fixtures/kitchen-sink.graphql'),
+  {
+    encoding: 'utf8',
+  },
+);
+const githubSchema = fs.readFileSync(
+  path.resolve(__dirname, './fixtures/github.graphql'),
+  {
+    encoding: 'utf8',
+  },
+);
+const githubGraphqlJsSchema = fs.readFileSync(
+  path.resolve(__dirname, './fixtures/github-graphqljs.graphql'),
```

Comment:
can you clarify the difference between these last two schemas? when I get a chance I could manually diff the two files, but just looking at both SDL files and filesizes i don't see the difference between them

Reply:
The difference is the file size only. We are performing two tests here:
1) parsing whole schema at once.
2) parsing line by line

For the 1st test we are testing with short and big files. For the 2nd we are only testing with the short one.

Reply:
![image](https://user-images.githubusercontent.com/1368727/89052052-1fdb4000-d323-11ea-8cba-d17044adc798.png)
maybe i'm mistaken, but both of these files look to be the same size? (kb on left)

Reply:
Oh, yes thanks. I missed it. Will fix.